datajunction-server/datajunction_server/api/materializations.py

```diff
@@ -520,3 +521,55 @@ async def run_materialization_backfill(
     )
     await session.commit()
     return materialization_output
+
+
+@router.get(
+    "/nodes/{node_name}/availability/",
+    response_model=List[AvailabilityStateInfo],
+    status_code=200,
+    name="List All Availability States for a Node",
+)
+async def list_node_availability_states(
+    node_name: str,
+    *,
+    session: AsyncSession = Depends(get_session),
+) -> List[AvailabilityStateInfo]:
+    """
+    Retrieve all availability states for a given node across all revisions.
+    """
+    # Get all revisions with their availability states
+    node = await Node.get_by_name(
+        session,
+        node_name,
+        options=[
```

Comment:
Should this endpoint optionally support only listing availability for the latest node revision?

Reply:
I didn't have a need for this call yet, but it would be easy to change when needed.